### PR TITLE
Fix the output units of pixel_to_world

### LINKED ISSUE #2324: Map plotting does not work without WCSAxes
I have a data set 600x600 that represents electron densities and I'm trying to put it into a sunpy map like so: 
```python
header = {'cdelt1':20, 'cdelt2':20, 'telescop':'Electron Density Map'}
den_map = sunpy.map.Map(data, header)
den_map.peek()
```
`den_map.peek()` works fine but when I try to use `den_map.plot()` I get the following error that doesn't make much sense: 
```python
ValueError: Latitude angle(s) must be within -90 deg <= angle <= 90 deg, got -5990.0 deg
```
This was tested on an older version of sunpy and it works fine but doesn't work with 0.8.1. Any help would be much appreciated.

### PR DESCRIPTION
**Pull request by Cadair**
This correctly infers the output units of `WCS.wcs_pix2world` by copying the WCS object and asking it what the units are after the transformation.

closes #2324

### CODE REVIEW

File: sunpy/map/mapbase.py [hunk 1]
```diff
@@ -855,15 +855,17 @@ def pixel_to_world(self, x, y, origin=0):
...
 
-        # If the wcs is celestial it is output in degress
-        if self.wcs.is_celestial:
-            x = u.Quantity(x, u.deg)
-            y = u.Quantity(y, u.deg)
-        else:
-            x = u.Quantity(x, self.spatial_units[0])
-            y = u.Quantity(y, self.spatial_units[1])
+        # Hold the WCS instance here so we can inspect the output units after
+        # the pix2world call
```
Comment (nabobalis) [inline sunpy/map/mapbase.py]: This inspection is done outside of the function call?

File: sunpy/map/mapbase.py [hunk 2]
```diff
@@ -855,15 +855,17 @@ def pixel_to_world(self, x, y, origin=0):
...
+        temp_wcs = self.wcs
```
Comment (nabobalis) [inline sunpy/map/mapbase.py]: Basically we don't want to change the actual WCS instance?

Comment (Cadair) [reply sunpy/map/mapbase.py]: There is no 'actual wcs instance' it's a property that makes one on access.

File: sunpy/map/mapbase.py (hunk 1 above)
Comment (Cadair) [reply sunpy/map/mapbase.py]: No it's done on the line after the call.

File: sunpy/map/mapbase.py (hunk 2 above)
Comment (nabobalis) [reply sunpy/map/mapbase.py]: To-mato, tom-ato. 

So this way, you do not change the property?

Comment (Cadair) [reply sunpy/map/mapbase.py]: We need to do the world2pix call and then inspect cunit on the same instance. If we re-accessed `self.wcs` it wouldn't be the same instance.

Comment (nabobalis) [reply sunpy/map/mapbase.py]: fair enough

**Comment by nabobalis** [comment]
Online time out only. Offline test for this passes. Merging.

**Pull request merged**